### PR TITLE
add sentry config

### LINKED ISSUE #224: Implement bug tracking software
**This work should be done after #225 as there is a sentry sdk for expo managed projects**

Integrate Sentry for bug tracking

## To do:
- [x] Install Sentry SDK
- [x] Initialize DSN with public key **@ErikSin will provide once we are approved for a non profit account **
- [x] Add Sentry Expo Plugin
- [x] Add auth token as a secret key **@ErikSin will provide once we are approved for a non profit account**
- [x] Catch and log and uncaught exceptions

### Configuration
- debug mode should be turned on in development and preview builds, not production builds.

### LINKED ISSUE #226: Use Expo App Services for CI Phase 1
Configure project to be built of Expo App Services.

## To Do:
- [x] Follow set up [instructions](https://docs.expo.dev/build/setup/). 
- [x] Allow for builds of [APK's for internal use](https://docs.expo.dev/build-reference/apk/)
- [x] Add [Sentry Auth Token](#224 ) as a [secret](https://docs.expo.dev/build-reference/variables/#using-secrets-in-environment-variables)

### PR DESCRIPTION
**Pull request by bohdanprog**
In that PR we added
- Sentry package. 
- Sentry config.
- Added to EAS auth sentry token.
- enable debug mode when the app variant is `development` or `test`.
Closes #226 
Closes #224

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -94,7 +94,8 @@
     "uint8array-extras": "^0.5.0",
     "utm": "^1.1.1",
     "validate-color": "^2.2.4",
-    "zustand": "^4.4.6"
+    "zustand": "^4.4.6",
+    "@sentry/react-native": "~5.20.0"
```
Comment (ErikSin) [inline package.json:98]: can we just run an `npm i` to get this to be in the proper alphabetical order

**Review by ErikSin — COMMENTED** [review]
Looks good.

We obviously need to figure out what we are going to track, but ill create a follow up ticket

Should developers at DD be using a Sentry Auth Token? Or is that unnecessary if we are not sending info to sentry during our development builds?

**Comment by CDFN** [comment]
Sentry Auth Token is unnecessary for development. It is only used by CI to create build for an app in Sentry (so you can track issues for specific build). Hope it makes sense for you 😅

**Comment by ErikSin** [comment]
Sorry, I screwed this up when I forced push the main branch. I forgot to re-add the expo splash screen work. Andrew is working to put it back, but we found another small problem that is blocking that work. He is trying to fix those right now

Im going to wait to merge the expo splash screen PR (and the other PR blocking). Once those are merged, we can update this PR and it will contain just the sentry work. Otherwise I think there isn't much more work that needs to go into this.

I am going to create a couple follow up ticket:
1. We need to actually catch errors now and send them to sentry
2. We also need to implement sentry in the backend

**Review by ErikSin — APPROVED** [review]
Can you rebase from main and make sure the code is running before merging please!

**Comment by ErikSin** [comment]
Be careful when updating from main. I just introduced a new PR that changes the file structure and it touched over 60 files (sorry). So make sure that you are only introducing sentry changes when you merge.

File: app.json [hunk 2]
```diff
@@ -7,6 +7,14 @@
...
     "icon": "./assets/icon.png",
     "userInterfaceStyle": "light",
+    "splash": {
+      "image": "./assets/splash.png",
+      "backgroundColor": "#050F77",
+      /*resizeMode - native work only for android
+      https://docs.expo.dev/develop/user-interface/splash-screen/#custom-configuration-for-android-and-ios
+      **/
+      "resizeMode": "native"
+    },
```
Comment (ErikSin) [inline app.json:17]: Andrew and I were having troubles with the splash screen and determined that this code actually does belong in the android section below.

Comment (ErikSin) [reply app.json:17]: According to the expo docs, the [splash section](https://docs.expo.dev/versions/latest/config/app/#splash) is for standalone apps

And the [splash/android](https://docs.expo.dev/versions/latest/config/app/#splash-2) is for managed android apps

**Pull request merged**